src/components/Heading/Heading.tsx

```diff
@@ -195,6 +195,11 @@ export const Heading: ChakraComponent<
               ...styles.base,
               ...wrapperStyles,
             };
+      /** When overline or subtitle is passed, the ...rest props should not be
+       * applied to the <h> element and should only be applied to the <hgroup>
+       * element. The conditional statement below is written such that when
+       * neither of those props are passed, ...rest is a applied. */
+      const headingRest = !overline && !subtitle && { ...rest };
```

Comment:
If it's either overline or subtitle, then change the first `&&` to `||`

Reply:
This should not be an `or` statement. `...rest` should only be applied here when both props are not passed.